src/frontend/screens/GpsModal.tsx

```diff
@@ -0,0 +1,183 @@
+import {useLocation} from '../hooks/useLocation';
+import {useLastKnownLocation} from '../hooks/useLastSavedLocation';
+import {useLocationProviderStatus} from '../hooks/useLocationProviderStatus';
+import {usePersistedSettings} from '../hooks/persistedState/usePersistedSettings';
+import {GPS_MODAL_TEXT, WHITE} from '../lib/styles';
+import {CustomHeaderLeft} from '../sharedComponents/CustomHeaderLeft';
+import {DateDistance} from '../sharedComponents/DateDistance';
+import {FormattedCoords} from '../sharedComponents/FormattedData';
+import {Text} from '../sharedComponents/Text';
+
+const m = defineMessages({
+  gpsHeader: {
+    id: 'screens.GpsModal.gpsHeader',
+    defaultMessage: 'Current GPS Location',
+    description: 'Header for GPS screen',
+  },
+  lastUpdate: {
+    id: 'screens.GpsModal.lastUpdate',
+    defaultMessage: 'Last update',
+    description: 'Section title for time of last GPS update',
+  },
+  utm: {
+    id: 'screens.GpsModal.locationUTM',
+    defaultMessage: 'Coordinates UTM',
+    description: 'Section title for UTM coordinates',
+  },
+  dms: {
+    id: 'screens.GpsModal.locationDMS',
+    defaultMessage: 'Coordinates DMS',
+    description: 'Section title for DMS coordinates',
+  },
+  dd: {
+    id: 'screens.GpsModal.locationDD',
+    defaultMessage: 'Coordinates Decimal Degrees',
+    description: 'Section title for DD coordinates',
+  },
+  details: {
+    id: 'screens.GpsModal.details',
+    defaultMessage: 'Details',
+    description: 'Section title for details about current position',
+  },
+  yes: {
+    id: 'screens.GpsModal.yes',
+    defaultMessage: 'Yes',
+    description: 'if a location sensor is active yes/no',
+  },
+  no: {
+    id: 'screens.GpsModal.no',
+    defaultMessage: 'No',
+    description: 'if a location sensor is active yes/no',
+  },
+  locationSensors: {
+    id: 'screens.GpsModal.locationSensors',
+    defaultMessage: 'Sensor Status',
+    description: 'Heading for section about location sensor status',
+  },
+});
+
+const GpsModalRow = ({label, value}: {label: string; value: string}) => (
+  <View style={styles.row}>
+    <Text style={styles.rowLabel}>{label}</Text>
+    <Text style={styles.rowValue}>{value}</Text>
+  </View>
+);
+
+export const GpsModal = () => {
+  const {location} = useLocation({maxDistanceInterval: 15});
+  const lastKnownLocationQuery = useLastKnownLocation();
+  const provider = useLocationProviderStatus();
+  const {coordinateFormat} = usePersistedSettings();
+  const {formatMessage: t} = useIntl();
+
+  const locationTimestamp =
+    location?.timestamp || lastKnownLocationQuery.data?.timestamp;
+
+  return (
+    <ScrollView style={styles.container} testID="gpsScreenScrollView">
+      <View style={styles.infoArea}>
+        <Text style={styles.sectionTitle}>
+          <FormattedMessage {...m.lastUpdate} />
+        </Text>
+        <DateDistance
+          style={styles.rowValue}
+          date={locationTimestamp ? new Date(locationTimestamp) : new Date()}
```

Comment:
noting that there's a notable difference in implementation for this between Mapeo Mobile and here

https://github.com/digidem/mapeo-mobile/blob/0c0ebbb9ef2261e21cd1d1c8bd5ab2fe42017ea3/src/frontend/screens/GpsModal.tsx#L86

https://github.com/digidem/mapeo-mobile/blob/0c0ebbb9ef2261e21cd1d1c8bd5ab2fe42017ea3/src/frontend/screens/GpsModal.tsx#L135-L140

Attempting to directly copy over the implementation caused infinite render loops when turning off location services. Seems to be something about `new Date('None')`, which returns `Invalid Date` (which is confusingly a `Date` type...) and that not playing nicely with React Intl's `FormattedRelativeTime` component (used in `DateDistance`).